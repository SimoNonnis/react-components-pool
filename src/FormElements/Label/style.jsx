import styled from 'styled-components';

export const LabelStyled = styled.label`
  color: ${({ theme }) => theme.color.veryDarkBlue};
  font-family: ${({ theme }) => theme.base.fontFamily};
  ${({ theme, error }) => error && `color: ${theme.color.red}`};
  ${({ theme, disabled }) => disabled && `color: ${theme.color.lightGrey2}`};
  margin-bottom: 7px;
  display: block;
  font-size: ${({ theme }) => theme.base.fontSize};
  font-weight: 600;
`;
LabelStyled.displayName = 'LabelStyled';

export const Asterisk = styled.span`
  color: ${({ theme, disabled }) => (disabled ? theme.color.lightGrey2 : theme.color.red)};
  display: inline-block;
  margin-left: 3px;
`;
Asterisk.displayName = 'Asterisk';
