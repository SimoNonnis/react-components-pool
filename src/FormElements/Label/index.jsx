import React from 'react';
import PropTypes from 'prop-types';
import { LabelStyled, Asterisk } from './style';

const propTypes = {
  error: PropTypes.bool,
  className: PropTypes.string,
  children: PropTypes.oneOfType([PropTypes.func, PropTypes.node]),
  htmlFor: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  required: PropTypes.bool,
};
const defaultProps = {
  error: false,
  className: null,
  children: null,
  disabled: false,
  required: false,
};

const Label = ({ disabled, error, className, children, htmlFor, required }) => {
  if (!children) return null;

  const asterisk = disabled || (required && <Asterisk>*</Asterisk>);

  return (
    <LabelStyled htmlFor={htmlFor} disabled={disabled} error={error} className={className}>
      {children}
      {asterisk}
    </LabelStyled>
  );
};

Label.propTypes = propTypes;
Label.defaultProps = defaultProps;

export default Label;
