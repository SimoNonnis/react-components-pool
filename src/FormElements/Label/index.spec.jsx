/* eslint-env jest */
import React from 'react';

import 'jest-styled-components';

import Label from './';

describe('Test <Label />', () => {
  it('should render Label correctly with default props', () => {
    const wrapper = shallow(<Label htmlFor="id" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render Label correctly with custom props', () => {
    const wrapper = shallow(
      <Label error className="className" htmlFor="id" required>
        Label Name
      </Label>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('should render Label correctly with custom props', () => {
    const wrapper = shallow(
      <Label error className="className" htmlFor="id" required disabled>
        Label Name
      </Label>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
