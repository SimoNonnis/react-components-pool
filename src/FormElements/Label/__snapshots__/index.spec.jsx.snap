// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Label /> should render Label correctly with custom props 1`] = `
<LabelStyled
  className="className"
  disabled={false}
  error={true}
  htmlFor="id"
>
  Label Name
  <Asterisk>
    *
  </Asterisk>
</LabelStyled>
`;

exports[`Test <Label /> should render Label correctly with custom props 2`] = `
<LabelStyled
  className="className"
  disabled={true}
  error={true}
  htmlFor="id"
>
  Label Name
</LabelStyled>
`;

exports[`Test <Label /> should render Label correctly with default props 1`] = `""`;
